### PR TITLE
{bio}[foss/2019b] HADDOCK v2.4-2020-09

### PR DESCRIPTION
**Pull request by jordiblasco**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    ('Boost', '1.71.0'),
+    ('CNS', '1.31-linux-x86_64', '', True)
+]
+
+#extract_sources = True
+install_type = 'merge'
+# HADDOCK interacts with the RMS and needs to be configured before installing it.
+# Modify the following parameters to set up this software for your cluster.
+# More information here: https://www.bonvinlab.org/software/haddock2.4/installation/#configuring-haddock
+preinstall_cmd = "echo 'set CNSTMP=$EBROOTCNS/cns_solve-1.31-UU-Linux64.exe' > haddock.conf && "
```
Comment (jordiblasco) [inline easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb:42]: This software relies on this particular release of CNS (Binary file). Not sure what is the best way to handle that in the easyconfig.

File: easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+sources = ['haddock%(version)s.tgz']
+checksums = ['334c45a3eb45473a9ac91f1b1d5687ef60aa4c6ad7b4064859c13c7941cd77ad']
+
+dependencies = [
+    ('libzeep', '3.0.5'),
+    ('bzip2', '1.0.8'),
+    ('zlib', '1.2.11'),
+    ('XZ', '5.2.4'),
+    ('Boost', '1.71.0'),
+    ('CNS', '1.31-linux-x86_64', '', True)
```
Comment (Micket) [inline easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb:34]:
```suggestion
    ('CNS', '1.31', '-linux-x86_64', True)
```
i think we should do it like this

File: easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb (hunk 1 above)
Comment (Micket) [reply easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb:42]: The versionnumber can be introduced as
```
local_cns_version = '1.31'
...
    ('CNS', local_cns_version, ...)
...
preinstall_cmd = "echo 'set CNSTMP=$EBROOTCNS/cns_solve-%s-UU-Linux64.exe' % local_cns_version
```

File: easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#extract_sources = True
+install_type = 'merge'
+# HADDOCK interacts with the RMS and needs to be configured before installing it.
+# Modify the following parameters to set up this software for your cluster.
+# More information here: https://www.bonvinlab.org/software/haddock2.4/installation/#configuring-haddock
+preinstall_cmd = "echo 'set CNSTMP=$EBROOTCNS/cns_solve-1.31-UU-Linux64.exe' > haddock.conf && "
+preinstall_cmd += "echo 'set QUEUETMP=/bin/csh' >> haddock.conf && "
+preinstall_cmd += "echo 'set NUMJOB=4' >> haddock.conf && "
+preinstall_cmd += "echo 'set QUEUESUB=QueueSubmit_concat.py' >> haddock.conf && "
+preinstall_cmd += "./install.csh haddock.conf"
```
Comment (Micket) [inline easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb:46]: hm, these don't look very general, doesn't seem like anything one would want to hardcode

**Review by Micket — CHANGES_REQUESTED** [review]

**Comment by verdurin** [comment]
@jordiblasco Any thoughts on the suggestions from @Micket ?

File: easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb (hunk 3 above)
Comment (jordiblasco) [reply easybuild/easyconfigs/h/HADDOCK/HADDOCK-2.4-2020-09-foss-2019b.eb:47]: I know it's not very general but I can't find a better option. 
The configuration file needs to be created and populated before the installation.
I'm open to suggestions.

**Review by jordiblasco — COMMENTED** [review]
Incorporated @Micket suggestions.

**Comment by jordiblasco** [comment]
Dependency is broken due to lack of CNS, which is available in #PR11688 https://github.com/easybuilders/easybuild-easyconfigs/pull/11688

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 11689` and update the PR to use `SYSTEM` instead.

**Comment by boegel** [comment]
closing this since `foss/2019b` is no longer supported, see https://docs.easybuild.io/policies/toolchains

**Pull request closed without merging**